runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -313,6 +313,15 @@ public synchronized ILogData previous() {
 
         // If still null, we're done.
         if (prevAddress == null) {
+            if (context.checkpointSuccessId == null) {
+                // The stream hasn't been checkpointed and we need to
```

Comment:
please add logs, you can even transform your comments into log messages

Reply:
done.